x-pack/plugins/lens/public/datatable_visualization/components/toolbar.tsx

```diff
@@ -11,18 +11,28 @@ import { EuiFlexGroup, EuiFormRow, EuiSwitch } from '@elastic/eui';
 import { ToolbarPopover } from '../../shared_components';
 import type { VisualizationToolbarProps } from '../../types';
 import type { DatatableVisualizationState } from '../visualization';
+import { DEFAULT_PAGE_SIZE } from './table_basic';
 
 export function DataTableToolbar(props: VisualizationToolbarProps<DatatableVisualizationState>) {
   const { state, setState } = props;
 
-  const onChange = useCallback(() => {
+  const onToggleFitRow = useCallback(() => {
     const current = state.fitRowToContent ?? false;
     setState({
       ...state,
       fitRowToContent: !current,
     });
   }, [setState, state]);
 
+  const onTogglePagination = useCallback(() => {
+    const current = state.paging ?? { size: DEFAULT_PAGE_SIZE, enabled: false };
+
+    setState({
+      ...state,
+      paging: { ...current, enabled: !current.enabled },
```

Comment:
Mostly because we are currently working on exactly this in TSVB - should we reset the page size if the user is disabling pagination?

Reply:
Yeah, valid point about this particular setting not hurting anyone. I trust your judgement 👍

Reply:
I showed this to @MichaelMarcialis and he thinks having the app remember the previous page size setting is a nice thing. So, because of that and the reasons above, let's keep it for now.